engine/rpcserver.go

```diff
@@ -1213,3 +1220,291 @@ func (s *RPCServer) GetAuditEvent(ctx context.Context, r *gctrpc.GetAuditEventRe
+		return &gctrpc.GCTScriptQueryResponse{Status: "error"}, nil
+	}
+
+	if v, f := gctscript.AllVMs[UUID]; f {
+		resp.Status = "ok"
+		resp.Script = new(gctrpc.GCTScript)
+		resp.Script.Name = v.ShortName()
+		resp.Script.UUID = v.ID.String()
+		resp.Script.Path = v.Path
+		resp.Script.NextRun = v.NextRun.String()
+		data, err := v.Read()
+		if err != nil {
+			return nil, err
+		}
+		resp.Data = string(data)
+	}
+	return resp, nil
+}
+
+// GCTScriptExecute execute a script
+func (s *RPCServer) GCTScriptExecute(ctx context.Context, r *gctrpc.GCTScriptExecuteRequest) (*gctrpc.GCTScriptGenericResponse, error) {
+	if !gctscript.GCTScriptConfig.Enabled {
+		return &gctrpc.GCTScriptGenericResponse{Status: gctscript.ErrScriptingDisabled.Error()}, nil
+	}
+
+	if r.Script.Path == "" {
+		r.Script.Path = gctscript.ScriptPath
+	}
+
+	gctVM := gctscript.New()
+	if gctVM == nil {
+		return &gctrpc.GCTScriptGenericResponse{}, nil
+	}
+
+	script := filepath.Join(r.Script.Path, r.Script.Name)
+	err := gctVM.Load(script)
+	if err != nil {
+		return &gctrpc.GCTScriptGenericResponse{
+			Status: "error",
+			Data:   err.Error(),
+		}, nil
+	}
+
+	go gctVM.CompileAndRun()
+
+	return &gctrpc.GCTScriptGenericResponse{
+		Status: "ok",
+		Data:   gctVM.ShortName() + " (" + gctVM.ID.String() + ") executed",
+	}, nil
+}
+
+// GCTScriptStop terminate a running script
+func (s *RPCServer) GCTScriptStop(ctx context.Context, r *gctrpc.GCTScriptStopRequest) (*gctrpc.GCTScriptGenericResponse, error) {
+	if !gctscript.GCTScriptConfig.Enabled {
+		return &gctrpc.GCTScriptGenericResponse{Status: gctscript.ErrScriptingDisabled.Error()}, nil
+	}
+
+	UUID, err := uuid.FromString(r.Script.UUID)
+	if err != nil {
+		return &gctrpc.GCTScriptGenericResponse{Status: "error", Data: err.Error()}, nil
+	}
+
+	if v, f := gctscript.AllVMs[UUID]; f {
+		err = v.Shutdown()
+		status := " terminated"
+		if err != nil {
+			status = err.Error()
+		}
+		return &gctrpc.GCTScriptGenericResponse{Status: "ok", Data: v.ID.String() + status}, nil
+	}
+	return &gctrpc.GCTScriptGenericResponse{Status: "error", Data: "no running script found"}, nil
+}
+
+// GCTScriptUpload upload a new script to ScriptPath
+func (s *RPCServer) GCTScriptUpload(ctx context.Context, r *gctrpc.GCTScriptUploadRequest) (*gctrpc.GCTScriptGenericResponse, error) {
+	if !gctscript.GCTScriptConfig.Enabled {
+		return &gctrpc.GCTScriptGenericResponse{Status: gctscript.ErrScriptingDisabled.Error()}, nil
+	}
+
+	fPath := filepath.Join(gctscript.ScriptPath, r.ScriptName)
+	_, err := os.Stat(fPath)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return nil, err
+		}
+	} else {
+		if !r.Overwrite {
+			return nil, fmt.Errorf("%s script found and overwrite set to false", r.ScriptName)
+		} else if r.Overwrite {
+			f := filepath.Join(gctscript.ScriptPath, "version_history")
+			err = os.MkdirAll(f, 0770)
+			if err != nil {
+				return nil, err
+			}
+			renamedFile := filepath.Join(f, time.Now().Format(time.RFC3339)+" "+r.ScriptName)
```

Comment:
Looks like this path is broken on Windows:

```
2020/01/02 13:22:26 rpc error: code = Unknown desc = open C:\Users\BobMarley\AppData\Roaming\GoCryptoTrader\scripts\version_history\2020-01-02T13:22:26+11:00 test.gct: The filename, directory name, or volume label syntax is incorrect.
```

Reply:
![image](https://user-images.githubusercontent.com/4685270/71649915-ad8f8680-2d66-11ea-9d2d-2a92a0c0650f.png)
Can also change the " " to "-" as it's easier with pathing

Reply:
![image](https://user-images.githubusercontent.com/1223381/71650015-5a6a0380-2d67-11ea-8746-d44890856c10.png)

Just pushed a change that moves this to use unix timestamp (with nano seconds)

Reply:
40e0c91 fixes this! Pro stuff 
```
gctcli.exe gctscript upload --path=C:\GCTScripts\test.gct --overwrite=true
{
 "status": "ok",
 "data": "script C:\\Users\\BobMarley\\AppData\\Roaming\\GoCryptoTrader\\scripts\\test.gct written"
}
gctcli.exe gctscript list
{
 "scripts": [
  {
   "name": "C:\\Users\\BobMarley\\AppData\\Roaming\\GoCryptoTrader\\scripts\\pairs.gct"
  },
  {
   "name": "C:\\Users\\BobMarley\\AppData\\Roaming\\GoCryptoTrader\\scripts\\test.gct"
  },
  {
   "name": "C:\\Users\\BobMarley\\AppData\\Roaming\\GoCryptoTrader\\scripts\\version_history\\1577934179796331200-test.gct"
  },
  {
   "name": "C:\\Users\\BobMarley\\AppData\\Roaming\\GoCryptoTrader\\scripts\\version_history\\1577934223262482200-test.gct"
  }
 ]
}
gctcli.exe gctscript execute test.gct
{
 "status": "ok",
 "data": "test.gct (07885bf3-968f-4a35-b043-9d12a5c01043) executed"
}
```